Domain/Problem/src/main/java/tw/waterball/judgegirl/problem/domain/repositories/ProblemQueryParams.java

```diff
@@ -18,45 +18,67 @@
 
 import java.util.Optional;
 
+import static java.util.Optional.ofNullable;
+
 /**
  * @author - johnny850807@gmail.com (Waterball)
  */
 public class ProblemQueryParams {
-    public static final ProblemQueryParams NO_PARAMS = new ProblemQueryParams(null, null, false);
+    public static final ProblemQueryParams NO_PARAMS = new ProblemQueryParams(null, null, false, true, true);
 
     private final String[] tags;
 
     @Nullable
     private final Integer page;
 
-    private final boolean excludeArchive;
+    @Nullable
+    private final Boolean queryArchive;
 
-    private final boolean includeInvisibleProblems;
+    private final boolean queryVisible;
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean includeInvisibleProblems) {
-        this(tags, page, true, includeInvisibleProblems);
-    }
+    private final boolean queryInvisible;
 
-    public ProblemQueryParams(String[] tags, @Nullable Integer page, boolean excludeArchive, boolean includeInvisibleProblems) {
+    /**
+     * To query the problems with the tags, then provide the tags of a not empty string array.
+     * To query the problems in the pages, then provide the page of an Integer, its default value is 0,
+     * and it will return pages of the problems with page size.
+     * To query the archived problems, then provide queryArchive of a boolean with true
+     * To query the non archived problems, then provide queryArchive of a boolean with false.
+     * To query the archived and non archived problems, then not provide queryArchive.
+     * To query the visible problems, then provide queryVisible of a boolean with true and queryInvisible of a boolean with false.
+     * To query the invisible problems, then provide queryInvisible of a boolean with true and queryVisible of a boolean with false.
```

Comment:
Thank you for complementing the comments. But I just come up with a better and concise way.

1. rename `queryVisible` to `includeVisibleProblems`
2. rename `queryInvisible` to `includeInvisibleProblems`
3. rename `queryArchive ` to `archivedFlag`
4. remove all this javadoc, we don't need it anymore

So that we will have the more readable code as follows in the repository:
```java
params.archivedFlag()
            .ifPresent(archivedFlag -> query.addCriteria(where("archived").is(archivedFlag)));

boolean includeVisibleProblems = params.includeVisibleProblems();
 boolean includeVisibleProblems = params.includeInvisibleProblems();
if (includeVisibleProblems && !includeInvisibleProblems) {
        query.addCriteria(where("visible").is(true));
 } else if (!includeVisibleProblems && includeInvisibleProblems) {
        query.addCriteria(where("visible").is(false));
 }
```